components/pkg-aci/bin/hab-pkg-aci.sh

```diff
@@ -102,44 +102,48 @@ find_system_commands() {
 
 package_name_for() {
   local pkg="$1"
-  echo $(echo $pkg | cut -d "/" -f 2)
+  echo "$pkg" | cut -d "/" -f 2
 }
 
 package_exposes() {
   local pkg="$1"
-  local expose_file=$(find $ACI_CONTEXT/rootfs/$HAB_ROOT_PATH/pkgs/$pkg -name EXPOSES)
+  local expose_file
+  expose_file=$(find "$ACI_CONTEXT"/rootfs/"$HAB_ROOT_PATH"/pkgs/"$pkg" -name EXPOSES)
   if [ -f "$expose_file" ]; then
-    cat $expose_file
+    cat "$expose_file"
   fi
 }
 
 package_version_tag() {
   local pkg="$1"
-  local ident_file=$(find $ACI_CONTEXT/rootfs/$HAB_ROOT_PATH/pkgs/$pkg -name IDENT)
-  cat $ident_file | awk 'BEGIN { FS = "/" }; { print $1 "-" $2 "-" $3 "-" $4 }'
+  local ident_file
+  ident_file=$(find "$ACI_CONTEXT"/rootfs/"$HAB_ROOT_PATH"/pkgs/"$pkg" -name IDENT)
+  awk 'BEGIN { FS = "/" }; { print $1 "-" $2 "-" $3 "-" $4 }' < "$ident_file"
 }
 
 build_aci() {
   ACI_CONTEXT="$($_mktemp_cmd -t -d "${program}-XXXX")"
   BUILD_DIR="$($_mktemp_cmd -t -d "${program}-XXXX")"
   pushd "$BUILD_DIR" > /dev/null
-  env PKGS="$@" NO_MOUNT=1 hab-studio -r $ACI_CONTEXT/rootfs -t baseimage new
-  local pkg_name=$(package_name_for $1)
-  local version_tag=$(package_version_tag $1)
-  echo "$1" > $ACI_CONTEXT/rootfs/.hab_pkg
-  $ACBUILD begin $ACI_CONTEXT/rootfs
-  $ACBUILD set-name $1
+  env PKGS="$*" NO_MOUNT=1 hab-studio -r "$ACI_CONTEXT"/rootfs -t baseimage new
```

Comment:
```
In /Volumes/vm/habitat/components/pkg-aci/bin/hab-pkg-aci.sh line 128:
  env PKGS="$@" NO_MOUNT=1 hab-studio -r "$ACI_CONTEXT"/rootfs -t baseimage new
            ^-- SC2145: Argument mixes string and array. Use * or separate argument.
```
I'm a bit confused here. We definitely don't want to be using `"$@"`, since that would break multiple arguments into separate tokens, making `build_aci foo bar` run something like
```
env PKGS=foo bar NO_MOUNT=1…
```
which will lead to a `env: ‘bar’: No such file or directory`-type error. On the other hand, `"@*"` results in the argument list being expanded to a single string, joined by the first char in `IFS` (usually a space):
```
env PKGS='foo bar' NO_MOUNT=1…
```
Which is a bit odd as env vars don't usually contain spaces, but it should probably work. What I don't see is how this would have ever worked before with multiple arguments to `build_aci`.

Reply:
I'd be surprised if anyone's ever run it that way, to be honest. I know there was some code in other exporters that suggested the ability to package multiple Habitat packages into one exported thing, but I don't think it was ever really fleshed out (e.g., you could add package X and Y, but the resulting output would only be named "X", giving no indication that anything else was baked in as well). 

This may be a vestige of that?